src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs

```diff
@@ -1088,6 +1082,278 @@ await using (var context = InMemoryApplicationDbContext(contextId))
+
+            await using (var context = InMemoryApplicationDbContext(contextId))
+            {
+                var releaseService = BuildReleaseService(contentDbContext: context,
+                    contentService: contentService.Object,
+                    releaseFileService: releaseFileService.Object);
+
+                var result = await releaseService
+                    .UpdateReleaseStatus(
+                        release.Id,
+                        new ReleaseStatusUpdateViewModel
+                        {
+                            ApprovalStatus = ReleaseApprovalStatus.Approved,
+                            LatestInternalReleaseNote = "Test note",
+                            PublishMethod = PublishMethod.Scheduled,
+                            NextReleaseDate = new PartialDate {Month="12", Year="2000"}
+                        }
+                    );
+
+                Assert.True(result.IsLeft);
+
+                AssertValidationProblem(result.Left, ApprovedReleaseMustHavePublishScheduledDate);
+            }
+
+            MockUtils.VerifyAllMocks(contentService, releaseFileService);
+        }
+
+        [Fact]
+        public async Task UpdateReleaseStatus_Approved_FailsChangingToDraft()
+        {
+            var release = new Release
+            {
+                Type = new ReleaseType
+                {
+                    Title = "Ad Hoc"
+                },
+                Publication = new Publication
+                {
+                    Title = "Old publication",
+                },
+                ReleaseName = "2030",
+                Slug = "2030",
+                Published = DateTime.Now,
+                PublishScheduled = DateTime.UtcNow,
+                Version = 0,
+            };
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await using (var context = InMemoryApplicationDbContext(contextId))
+            {
+                await context.AddAsync(release);
+                await context.SaveChangesAsync();
+            }
+
+            var contentService = new Mock<IContentService>(MockBehavior.Strict);
+            var releaseFileService = new Mock<IReleaseFileService>(MockBehavior.Strict);
+
+            await using (var context = InMemoryApplicationDbContext(contextId))
+            {
+                var releaseService = BuildReleaseService(contentDbContext: context,
+                    contentService: contentService.Object,
+                    releaseFileService: releaseFileService.Object);
+
+                var result = await releaseService
+                    .UpdateReleaseStatus(
+                        release.Id,
+                        new ReleaseStatusUpdateViewModel
+                        {
+                            ApprovalStatus = ReleaseApprovalStatus.Draft,
+                        }
+                    );
+
+                Assert.True(result.IsLeft);
```

Comment:
In the same way we have `var xxx = result.AssertRight()`, it'd probably be nice to have an `result.AssertValidationProblems(prob1, prob2, prob3)` that does both the Left check, and also does 1 or more `AssertValidationProblem` checks as well.

Not suggesting as part of this PR though!